pkg/frame/z16.go

```diff
@@ -0,0 +1,36 @@
+package frame
+
+import (
+	"encoding/binary"
+	"fmt"
+	"image"
+	"image/color"
+)
+
+func decodeZ16(frame []byte, width, height int) (image.Image, func(), error) {
+	expectedSize := 2 * (width * height)
+	if 2*(width*height) != len(frame) {
+		return nil, func() {}, fmt.Errorf("frame length (%d) not expected size (%d)", len(frame), expectedSize)
+	}
+	img := image.NewGray16(image.Rect(0, 0, width, height))
+	/*
+		v4l specifies images in terms of series of lines which is perplexing because the
+		example in https://www.kernel.org/doc/html/v5.9/userspace-api/media/v4l/pixfmt-z16.html
+		seems to swap the subscript of each depth value.
+
+		Clear example:
+		Width: 3, Height: 4
+		[Z_low(x_0,y_0), Z_high(x_0,y_0), Z_low(x_1,y_0), Z_high(x_1,y_0), Z_low(x_2,y_0), Z_high(x_2,y_0),
+		 Z_low(x_0,y_1), Z_high(x_0,y_1), Z_low(x_1,y_1), Z_high(x_1,y_1), Z_low(x_2,y_1), Z_high(x_2,y_1),
+		 Z_low(x_0,y_2), Z_high(x_0,y_2), Z_low(x_1,y_2), Z_high(x_1,y_2), Z_low(x_2,y_2), Z_high(x_2,y_2),
+		 Z_low(x_0,y_3), Z_high(x_0,y_3), Z_low(x_1,y_3), Z_high(x_1,y_3), Z_low(x_2,y_3)] Z_high(x_2,y_3),
```

Comment:
It would be nice to have a simple test to decode a sample binary.
(tests for yuv formats: https://github.com/pion/mediadevices/blob/master/pkg/frame/yuv_test.go

Reply:
Done!